TheAlienThatEatsTheCarrot/Model/BoardObject/BoardObject.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  GameObject.swift
+//  TheAlienThatEatsTheCarrot
+//
+//  Created by Sun Xinyu on 12/3/24.
+//
+
+import CoreGraphics
+
+// TODO: merged to GameObject in Component of ECS
+protocol BoardObject {
+    var position: CGPoint { get set }
+    var width: CGFloat { get set }
+    var height: CGFloat { get set }
+    var imageName: String? { get set }
+
+    func move(to newPosition: CGPoint)
```

Comment:
do we need functions for board objects, or should they purely store data for persistence? I am thinking functions are related to gameplay

Reply:
yes I do agree that this is unnecessary eventually. I have placed it here to facilitate for level designer and plan to remove it when ECS is ready.